pingpong/ud_pingpong.c

```diff
@@ -519,7 +519,11 @@ static int run(void)
 		for (i = 0; i < TEST_CNT; i++) {
 			if (test_size[i].option > opts.size_option)
 				continue;
+
 			opts.transfer_size = test_size[i].size;
+			if (opts.transfer_size > buffer_size)
+				continue;
```

Comment:
If a transfer size is given, you could use that as input for fi_info:max_msg_size.